nck/readers/adobe_reader_2_0.py

```diff
@@ -143,21 +148,30 @@ def __init__(
         metric,
         start_date,
         end_date,
+        date_range,
     ):
-        self.adobe_client = AdobeClient(
-            client_id, client_secret, tech_account_id, org_id, private_key
-        )
+        self.adobe_client = AdobeClient(client_id, client_secret, tech_account_id, org_id, private_key)
         self.global_company_id = global_company_id
         self.report_suite_id = report_suite_id
         self.dimensions = list(dimension)
         self.metrics = list(metric)
         self.start_date = start_date
-        self.end_date = end_date + timedelta(days=1)
+        if end_date is not None:
+            self.end_date = end_date + timedelta(days=1)
+        else:
+            self.end_date = end_date
+        self.date_range = date_range
         self.ingestion_tracker = []
         self.node_values = {}
 
     def build_date_range(self):
-        return f"{self.start_date.strftime(DATEFORMAT)}/{self.end_date.strftime(DATEFORMAT)}"
+        if self.start_date is not None and self.end_date is not None and self.date_range is None:
```

Comment:
So do we raise an error in case of both start/end dates and date range are given by the user? What about checking if end-start = date range in case all parameters are given ?
Btw seems like the same error would occur if only one of the start/end date is None and the other one is not.

Reply:
Yes correct. This is documented in the issue #67. We do this to avoid conflicts between a date range and 2 dates. For example, if the user set a date start and a date stop but also a date range to LAST_7_DAYS, what would we do? We would have to give the priority to one parameter on the other but it would be implicit.

Your suggestion would work but it would make things more difficult imo, so I prefer letting things like that.

Reply:
Yes I agree, this is enough for now. The value/time ratio for the feature I mentioned is too low to consider it for now.